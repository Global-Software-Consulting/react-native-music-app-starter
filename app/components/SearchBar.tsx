import React, { useState } from 'react';
import {
  View,
  StyleSheet,
  TextInput,
  TouchableOpacity,
  FlatList,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome5';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { useTheme, Text } from 'react-native-paper';
import { useDispatch, useSelector } from 'react-redux';
import { IPlayerState } from '../models/reducers/player';
import PlaylistSongsCard from '../components/Playlist/PlaylistSongs/PlaylistSongsCard'
interface TitleProps {
  // title: string,
  setUpdatedPlaylist?: any;
  placeholder?: any;
  Playlist?: any;
}
interface IPState {
  playerReducer: IPlayerState;

}
const SearchBar: React.FC<TitleProps> = (props): JSX.Element => {
  const styles = useStyles();
  const [query, setQuery] = useState('');
  const [mode, setMode] = useState(false);
  const theme = useTheme();
  const playList = useSelector((state: IPState) => state.playerReducer.playList);

  const onChangeText = (query: string) => {
    setQuery(query)
    const list = playList?.filter((val:any) => {
      return val.name.toLowerCase().includes(query.toLowerCase());
    });
    props.setUpdatedPlaylist(list);
  }
  return (

    <View style={styles.container}>
      <View style={styles.container2}>
        <Icon name={'search'} size={20} color={theme.colors.primary} />
      </View>

      {props.onFocus ? (
        <TouchableOpacity
          onPress={props.onFocus}
          style={{ flex: 4, justifyContent: 'center', padding: 5 }}>
          <Text style={styles.txt}>{props.placeholder}</Text>
        </TouchableOpacity>
      ) : (
        <View style={{ flex: 4, justifyContent: 'center' }}>
          <TextInput
            style={styles.txt}
            value={query}
            onChangeText={text => {
              onChangeText(text)
            }}
            placeholder={props.placeholder}
          />
        </View>
      )}
      
    </View>

    

  )
};

export const useStyles = () => {
  const theme = useTheme();
  const styles = StyleSheet.create({

    container: {
      height: hp('5%'),
      width: wp('85%'),
      flexDirection: 'row',
      paddingRight: 5,
      marginLeft: 15,
      borderRadius: 10,
      backgroundColor: theme.colors.background,
      shadowColor: "#000",
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowOpacity: 0.25,
      shadowRadius: 3.84,

      elevation: 5,
    },
    label: {
      fontSize: hp('3%'),
      fontWeight: 'bold',
      // color: theme.colors.accent,

    },

    container2: {
      backgroundColor: theme.colors.background,
      borderRadius: 10,
      justifyContent: 'center',
      padding: 10,
     
    },
    txt: {
      padding: 5,
      color: theme.colors.primary,
      fontSize: hp('2%'),
    },
    Searchtxt: {
      padding: 5,
      color: theme.colors.background,
      fontSize: hp('2%'),
    },
  });
  return styles;
}
export default SearchBar;