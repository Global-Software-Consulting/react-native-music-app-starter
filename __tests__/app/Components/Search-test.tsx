/**
 * @format
 */
import 'react-native';
import React from 'react';
import SearchBar from '../../../app/components/SearchBar';
// Note: test renderer must be required after react-native.
import { fireEvent, render } from '@testing-library/react-native';


jest.mock("react-native-responsive-screen", () => {
    return {
        heightPercentageToDP: jest.fn(),
        widthPercentageToDP: jest.fn(),

    }
});

jest.mock('@react-navigation/native', () => {
    return {
        useIsFocused: jest.fn(),
        useNavigation: jest.fn(),
        useRoute: jest.fn(),
    }
});
jest.mock('react-i18next', () => {
    return {
        useTranslation: () => {
            return {
                t: jest.fn(),
            };
        },
    };
});

jest.mock('react-native-iphone-x-helper', () => ({
    getStatusBarHeight: jest.fn(),
    getBottomSpace: jest.fn(),
}));
jest.mock('react-redux', () => ({
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
}));
describe("group tesing goes here", () => {
    test('adding songs', () => {
        const { getByTestId } = render(<SearchBar setUpdatedPlaylist={() => { }} />);
        // fireEvent.press(getByTestId('changeTest'));
        // fireEvent(getByPlaceholderText('changeTest'), 'onChangeText', 'ab');
        fireEvent.changeText(getByTestId('changeTest'));
    });
    test('adding songs', () => {
        const { getByTestId } = render(<SearchBar onFocus={() => { }} />);
        fireEvent.press(getByTestId('focus'));
    });

})

