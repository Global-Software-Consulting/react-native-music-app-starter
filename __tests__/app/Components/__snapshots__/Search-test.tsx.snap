// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group tesing goes here renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f6f6f6",
      "borderRadius": 10,
      "elevation": 5,
      "flexDirection": "row",
      "height": undefined,
      "marginLeft": 15,
      "marginTop": 10,
      "paddingRight": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
      "width": undefined,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#f6f6f6",
        "borderRadius": 10,
        "justifyContent": "center",
        "paddingLeft": 10,
      }
    }
  >
    <mockedDefaultExport
      color="#6200ee"
      name="search"
      size={15}
    />
  </View>
  <View
    style={
      Object {
        "flex": 4,
        "justifyContent": "center",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#6200ee",
          "fontSize": undefined,
          "padding": 5,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
